cage.c

```diff
@@ -588,7 +583,7 @@ main(int argc, char *argv[])
 	}
 #endif
 
-	if (!spawn_primary_client(&server, argv + optind, &pid, &sigchld_source)) {
+	if (optind < argc && !spawn_primary_client(&server, argv + optind, &pid, &sigchld_source)) {
```

Comment:
Is `cleanup_primary_client()` shoud also be conditional? Maybe checking `pid` not equal zero even if `waitpid()` can manage  such a value.

Reply:
Good catch, fixed.